xcode/Subconscious/Shared/Services/Noosphere/NoosphereService.swift

```diff
@@ -96,22 +94,24 @@ actor NoosphereService:
         self.sphereStorageURL = sphereStorageURL
         self.gatewayURL = gatewayURL
         self._sphereIdentity = sphereIdentity
-        self.logger = logger
         self._noosphereLogLevel = noosphereLogLevel
+        self.errorLoggingService = errorLoggingService
     }
     
     /// Create a default sphere for user and persist sphere details
     /// This creates, but does not save the sphere as default.
     /// - Returns: SphereReceipt
     func createSphere(ownerKeyName: String) async throws -> SphereReceipt {
-        try await self.noosphere().createSphere(
-            ownerKeyName: ownerKeyName
-        )
+        try await errorLoggingService.capturing {
```

Comment:
In all public methods that call to Noosphere or Sphere FFI wrappers, we wrap those calls with `errorLoggingService.capturing`. This will log any errors that happen as a side-effect, and then re-throw the error.